test/hotspot/jtreg/vmTestbase/nsk/jvmti/GetAllThreads/allthr001/allthr001.cpp

```diff
@@ -196,20 +225,33 @@ JNIEXPORT void checkInfo(JNIEnv *env, int ind) {
         return;
     }
 
+    // check unexpected threads
     for (i = 0; i < threadsCount; i++) {
-        if (!isThreadExpected(jvmti, threads[i])) {
-            num_unexpected++;
+        err = jvmti->GetThreadInfo(threads[i], &inf);
+        if (err != JVMTI_ERROR_NONE) {
+            printf("Failed to get thread info: %s (%d)\n",
+                   TranslateError(err), err);
+            result = STATUS_FAILED;
+            return;
+        }
+        if (printdump == JNI_TRUE) {
+            printf(" >>> %s", inf.name);
         }
+        for (j = 0, found = 0; j < thrInfo[ind].unexpected.cnt && !found; j++) {
```

Comment:
found = JNI_FALSE

Reply:
found is int, not jboolean.
I changed it to bool to be more C++-style